src/mig/modules/pkgprint/templates.go

```diff
@@ -0,0 +1,91 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Contributor: Aaron Meihm ameihm@mozilla.com [:alm]
+package pkgprint
+
+import (
+	"fmt"
+	"regexp"
+)
+
+type ppTemplate struct {
+	name        string
+	fingerprint fingerprint
+}
+
+type fingerprint struct {
+	filename     string
+	isRegexp     bool
+	pathFilter   string
+	contentMatch string
+	transform    func(string) (string, error)
+	forceRoot    string
+}
```

Comment:
This looks good, but what would extending it to netstat/memory fingerprints look like? Would we add more fields or should we regroup them into a smaller, more general, set of fields?

For example, there could be a `fpType` method that indicates `file`, `netstat` or `memory` and a `fpSignature` that contains a regex. Or is that too generic?